lucene/src/main/java/org/hibernate/hql/lucene/internal/ClassBasedLuceneQueryResolverDelegate.java

```diff
@@ -313,7 +313,7 @@ public void propertyPathCompleted(PropertyPath path) {
 
 	@Override
 	public void registerJoinAlias(Tree alias, PropertyPath path) {
-		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias ) ) {
+		if ( !path.getNodes().isEmpty() && !aliasToPropertyPath.containsKey( alias.getText() ) ) {
```

Comment:
mis-coordination between brain and finger? :) The issue is benign because only one join alias is  supported.